pyanaconda/modules/subscription/initialization.py

```diff
@@ -0,0 +1,90 @@
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+RHSM_SYSTEMD_UNIT_NAME = "rhsm.service"
+
+SERVICE_CHECK_INTERVAL = 0.1  # how often to check in s
+
+import time
+
+from dasbus.typing import get_variant, Str
+
+from pyanaconda.core import util
+
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.constants.services import RHSM
+from pyanaconda.modules.common.constants.objects import RHSM_CONFIG
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+class StartRHSMTask(Task):
+    """Task for starting the RHSM DBus service."""
+
+    def __init__(self):
+        super().__init__()
+        self._started_time = None
+
+    @property
+    def name(self):
+        return "Start RHSM DBus service"
+
+    def run(self):
+        """Start the RHSM DBus service.
+
+        And also some related tasks, such as setting RHSM log levels.
+        """
+        # record when the task was started
+        self._start_time = time.time()
+
+        # start the rhsm.service
+        # - this is blocking, but as we are effectively running in a thread
+        # it should not be an issue
+        # - if the return code is non-zero, raise an exception
+        rc = util.start_service(RHSM_SYSTEMD_UNIT_NAME)
+        if rc:
+            return False
+        # create a temporary proxy to set the log levels
+        rhsm_config_proxy = RHSM.get_proxy(RHSM_CONFIG)
+
+        # set RHSM log levels to debug
+        # - otherwise the RHSM log output is not usable for debugging subscription issues
+        log.debug("subscription: setting RHSM log level to DEBUG")
+        rhsm_config_proxy.Set("logging.default_log_level", get_variant(Str, "DEBUG"), "")
+        # all seems fine
+        return True
+
+    def is_service_available(self, timeout, check_interval=SERVICE_CHECK_INTERVAL):
+        """Return if RHSM service is available or wait if startup is ongoing."""
+        while self.is_running:
+            # wait up to defined timeout for the service to startup
+            time.sleep(check_interval)
+            # check if we should time out
+            if time.time() - self._start_time > timeout:
+                # This means the task is still running even after exceeding
+                # the define timeout and is likely stuck. We can't expect
+                # the RHSM DBus API be available in such a state, so return False,
+                # which will also terminate the while cycle.
+                log.debug("RHSM service failed to start in %f seconds", timeout)
+                return False
+        # If we got this far, the task has finished running. If the result is True
+        # it was able to successfully start the systemd unit and connect to the DBus API.
+        # If the result is False, then the service failed to start.
+        return self.get_result()
```

Comment:
I would handle the exception `NoResultError` just to be sure.

Reply:
Added.